.github/workflows/nightly.yml

```diff
@@ -0,0 +1,255 @@
+name: PL/Rust nightly builds
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+  pull_request:
+    branches:
+      - main
+      - develop
+  workflow_dispatch:
+
+# CODE REVIEW: Ensure this event setup is used and not on push/PR below!
+# CODE REVIEW: Ensure this event setup is used and not on push/PR below!
+# CODE REVIEW: Ensure this event setup is used and not on push/PR below!
+# on:
+#   schedule:
+#     - cron: '0 7 * * *'
+#   workflow_dispatch:
```

Comment:
I promise Ill change this before merging in. I am leaving it like this until it's been reviewed because any subsequent CI run that's needed in the future (pre-merging) should also re-run the nightly things to be sure.